src/odbc/sql2tds.c

```diff
@@ -149,6 +149,139 @@ _odbc_blob_free(TDSCOLUMN *col)
         TDS_ZERO_FREE(col->column_data);
 }
 
+static TDS_INT
+odbc_convert_table_row(TDS_DESC * apd, TDS_DESC * ipd,
+	SQLUSMALLINT ipar, SQLSMALLINT fParamType, SQLSMALLINT fCType, SQLSMALLINT fSqlType,
+	SQLULEN cbColDef, SQLSMALLINT ibScale, SQLPOINTER rgbValue, SQLLEN cbValueMax, SQLLEN FAR * pcbValue)
+{
+	int is_numeric = 0;
+	struct _drecord *drec;
+
+	if (fSqlType == SQL_DECIMAL || fSqlType == SQL_NUMERIC)
+		is_numeric = 1;
+
+	if (ipar > apd->header.sql_desc_count && desc_alloc_records(apd, ipar) != SQL_SUCCESS)
```

Comment:
`_SQLBindParameter` will do the validation for `ipar` when we bind the columns of the table, so we should be able to assume that it will contain a valid value.
https://github.com/FreeTDS/freetds/blob/3a16761a8e39d3b289fee0356ae7116be686713e/src/odbc/odbc.c#L1451